packages/cli/src/zosfiles/edit/Edit.handler.ts

```diff
@@ -0,0 +1,85 @@
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, GuiResult, IHandlerParameters,
+    ITaskWithStatus, ImperativeError, ProcessUtils, TaskStage, TextUtils } from "@zowe/imperative";
+import { IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
+import { ZosFilesBaseHandler } from "../ZosFilesBase.handler";
+import { EditUtilities as Utils, Prompt, EditFileType, LocalFile } from "../edit/Edit.utils";
+
+/**
+ * Handler to Edit USS or DS content locally
+ * @export
+ */
+export default class EditHandler extends ZosFilesBaseHandler {
+    public async processWithSession(commandParameters: IHandlerParameters, session: AbstractSession): Promise<IZosFilesResponse> {
+        // Setup
+        let lfFile = new LocalFile;
+        lfFile.guiAvail = ProcessUtils.isGuiAvailable() === GuiResult.GUI_AVAILABLE;
+        lfFile.fileName = commandParameters.arguments.file ?? commandParameters.arguments.dataSetName;
+        lfFile.tempPath = commandParameters.arguments.localFilePath = await Utils.buildTempPath(lfFile, commandParameters);
+        lfFile.fileType = commandParameters.positionals.includes('ds') ? EditFileType.ds : EditFileType.uss;
+
+        // Use or override stash (either way need to retrieve etag)
+        const stash: boolean = await Utils.checkForStash(lfFile.tempPath);
+        let useStash: boolean = false;
+        if (stash) {
+            useStash = await Utils.promptUser(Prompt.useStash);
+        }
+
+        // Download etag and possibly mf file to edit locally (if not using stash)
+        try{
+            const task: ITaskWithStatus = {
+                percentComplete: 10,
+                statusMessage: "Retrieving file",
+                stageName: TaskStage.IN_PROGRESS
+            };
+            commandParameters.response.progress.startBar({task});
+
+            // show a file comparision for the purpose of seeing the newer version of the remote mf file compared to your local edits
+            if (useStash && lfFile.guiAvail){
+                Utils.fileComparison(session, commandParameters);
+            }
+            lfFile = await Utils.localDownload(session, lfFile, useStash);
+
+            task.percentComplete = 70;
+            commandParameters.response.progress.endBar();
+        }catch(error){
+            if (error.causeErrors && error.causeErrors.code == 'ENOTFOUND'){
+                throw new ImperativeError({
+                    msg: TextUtils.chalk.red(`ENOTFOUND: Unable to connect to mainframe.`),
+                    causeErrors: error
+                });
+            }
+            throw new ImperativeError({
+                msg: TextUtils.chalk.red(`File not found on mainframe. Command terminated.`),
+                causeErrors: error
+            });
```

Comment:
For error handling, we want to be careful about assuming what an error means so that our error message isn't misleading. Also wrapping an error that may already be an ImperativeError in a new ImperativeError may obscure the original error message.

If we want to make the error message more useful for data sets that are not found, how about something like this that catches 404 errors to handle them specially, and for other error types simply rethrows them?
```typescript
if (error instanceof ImperativeError && error.errorCode === RestConstants.HTTP_STATUS_404) {
    throw new ImperativeError({
        msg: TextUtils.chalk.red(`File not found on mainframe. Command terminated.`),
        causeErrors: error
    });
} else {
    throw error;
}
```